src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubNotificationContext.java

```diff
@@ -145,14 +147,37 @@ public int hashCode() {
      * @since TODO
      */
     public String getDefaultContext(TaskListener listener) {
+        String jenkinshost = System.getenv("JENKINS_ID");
+        if (jenkinshost == null || jenkinshost.isEmpty() ) {
+            BufferedReader bf = null;
+            try {
+                Process process;
+                process = Runtime.getRuntime().exec("hostname");
```

Comment:
This is going to be a behaviour change for most systems I reckon. Also not a fan of a random env var being the setting that causes this. There should be a UI configuration for this IMHO.

Reply:
@res0nance you are right, it's behaviour change, and UI option for this would be great. Unfortunately I'm not the one who can do it. 
Nevertheless I believe this change would be useful for others too.

Reply:
@res0nance you are right, it's behaviour change, and UI option for this would be great. Unfortunately I'm not the one who can do it. 
Nevertheless I believe this change would be useful for others too.